pkg/controller/petset/pet.go

```diff
@@ -0,0 +1,267 @@
+package petset
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/client/record"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/golang/glog"
+)
+
+type petLifeCycleEvent string
+
+const (
+	syncPet              petLifeCycleEvent = "sync"
+	deletePet            petLifeCycleEvent = "delete"
+	updateRetries                          = 3
+	PetSetInitAnnotation                   = "pod.alpha.kubernetes.io/initialized"
+)
+
+// pcb is the control block used to transmit all updates about a single pet.
+type pcb struct {
```

Comment:
it's just a datastructure for passing around consolidated information about a pet more than what's in the api object. You might want to either examing the code for its use or wait till I add doc strings and remove the wip in the title.